File: test5/distinct_lines.py
#!/usr/bin/python3
import sys
from sys import argv

num = int(argv[1])
A=[]
count = 0
for i in sys.stdin:
    i = i.lower()
    L = i.split( )
    string = ' '.join(L)
    if string not in A:
        A.append(string)
    count += 1
    if len(A) == num :
        print("{} distinct lines seen after {} lines read.".format(num, count))
        break

if len(A) < num:
    print("End of input reached after {} lines read -  {} different lines not seen.".format(count,num))

File: test3/distinct_lines.pl
#!/usr/bin/perl -w

$num = $ARGV[0];
$count = 0;
$sum = 0;
while (<STDIN>){
	$IN = $_;
	$IN =~ s/ +//g;
	$sum += 1;
	if( exists($count_hash{lc($IN)})){
		next
	}
	else{
		$count_hash{lc($IN)} = 1;
		$count += 1;
	}
	if ($count == $num){
		print "$num distinct lines seen after $sum lines read.\n";
		exit;}}

if ($count < $num){
	print "End of input reached after $sum lines read - $num different lines not seen.\n";}
